internal/syncd/integrations/ecr/reconciler_test.go

```diff
@@ -0,0 +1,271 @@
+	mockRepoService.On("UpdateFile", fmt.Sprintf("Image Tag updated to: %s", inputImage.Tag), fakeReconciler.Branch, mock.Anything, []byte(expectedYaml)).Return(&github.RepositoryContentResponse{}, error(nil))
+
+	err := updateResource(fakeReconciler, inputImage, "bar")
+
+	assert.NoError(t, err)
+	mockRepoService.AssertExpectations(t)
+}
+
+func TestUpdateResourceDownloadFailure(t *testing.T) {
+	mockRepoService := new(MockRepoService)
+	fakeReconciler := &ImageReconciler{
+		Org:               "Wattpad",
+		RegistryChartPath: "foo/bar/resources",
+		Branch:            "oof",
+		RepoService:       mockRepoService,
+	}
+
+	inputImage := &internal.Image{
+		Registry:   "723255503624.dkr.ecr.us-east-1.amazonaws.com",
+		Repository: "bar",
+		Tag:        "78bc9ccf64eb838c6a0e0492ded722274925e2bd",
+	}
+
+	mockRepoService.On("GetFile", fakeReconciler.Branch, filepath.Join(fakeReconciler.RegistryChartPath, inputImage.Repository+".yaml")).Return("", fmt.Errorf("some error"))
+	err := updateResource(fakeReconciler, inputImage, "bar")
+	assert.Error(t, err)
+	mockRepoService.AssertExpectations(t)
+}
+
+func TestUpdateResourceUploadFailure(t *testing.T) {
+	mockRepoService := new(MockRepoService)
+	fakeReconciler := &ImageReconciler{
+		Org:               "Wattpad",
+		RegistryChartPath: "foo/bar/resources",
+		Branch:            "oof",
+		RepoService:       mockRepoService,
+	}
+
+	inputImage := &internal.Image{
+		Registry:   "723255503624.dkr.ecr.us-east-1.amazonaws.com",
+		Repository: "bar",
+		Tag:        "78bc9ccf64eb838c6a0e0492ded722274925e2bd",
+	}
+	path := filepath.Join(fakeReconciler.RegistryChartPath, inputImage.Repository+".yaml")
+	mockRepoService.On("GetFile", fakeReconciler.Branch, path).Return(inputYaml, error(nil))
+	mockRepoService.On("UpdateFile", fmt.Sprintf("Image Tag updated to: %s", inputImage.Tag), fakeReconciler.Branch, path, []byte(expectedYaml)).Return((*github.RepositoryContentResponse)(nil), fmt.Errorf("some upload error"))
+
+	err := updateResource(fakeReconciler, inputImage, "bar")
+
+	assert.Error(t, err)
+	mockRepoService.AssertExpectations(t)
+}
+
+func TestUpdateResourceInvalidYaml(t *testing.T) {
+	mockRepoService := new(MockRepoService)
+	fakeReconciler := &ImageReconciler{
+		Org:               "Wattpad",
+		RegistryChartPath: "foo/bar/resources",
+		Branch:            "oof",
+		RepoService:       mockRepoService,
+	}
+
+	inputImage := &internal.Image{
+		Registry:   "723255503624.dkr.ecr.us-east-1.amazonaws.com",
+		Repository: "bar",
+		Tag:        "78bc9ccf64eb838c6a0e0492ded722274925e2bd",
+	}
+
+	mockRepoService.On("GetFile", fakeReconciler.Branch, filepath.Join(fakeReconciler.RegistryChartPath, inputImage.Repository+".yaml")).Return("some malformed yaml", error(nil))
+	err := updateResource(fakeReconciler, inputImage, "bar")
+	assert.Error(t, err)
+	mockRepoService.AssertExpectations(t)
+}
+
+func TestNewReconciler(t *testing.T) {
```

Comment:
I don't think this test adds much value. If we want code coverage for the `NewReconciler` func, we should be calling it in the other test cases instead of constructing `ImageReconciler` with composite-literal syntax.

Reply:
Removed it and called `NewReconciler` in the other test cases.